go/client/cmd_chat_api.go

```diff
@@ -153,21 +153,37 @@ func (c *CmdChatAPI) ListV1(ctx context.Context) Reply {
 	rlimits = append(rlimits, inbox.RateLimits...)
 
 	var cl ChatList
-	cl.Conversations = make([]ConvSummary, len(inbox.Inbox.Conversations))
-	for i, conv := range inbox.Inbox.Conversations {
-		if len(conv.MaxMsgs) == 0 {
-			return c.errReply(fmt.Errorf("conversation %d had no max msgs", conv.Metadata.ConversationID))
-		}
-		tlf := conv.MaxMsgs[0].ClientHeader.TlfName
-		pub := conv.MaxMsgs[0].ClientHeader.TlfPublic
-		cl.Conversations[i] = ConvSummary{
-			ID: conv.Metadata.ConversationID,
-			Channel: ChatChannel{
-				Name:      tlf,
-				Public:    pub,
-				TopicType: strings.ToLower(conv.Metadata.IdTriple.TopicType.String()),
-			},
+	cl.Conversations = make([]ConvSummary, len(inbox.Conversations))
+	for i, conv := range inbox.Conversations {
+	maxMessagesLoop:
+		for _, msg := range conv.MaxMessages {
+			if msg.Message != nil {
+				var v1 chat1.MessagePlaintextV1
+				version, err := msg.Message.MessagePlaintext.Version()
+				if err != nil {
+					return c.errReply(err)
+				}
+				switch version {
+				case chat1.MessagePlaintextVersion_V1:
+					v1 = msg.Message.MessagePlaintext.V1()
+				default:
+					return c.errReply(libkb.NewChatMessageVersionError(version))
+				}
+
+				tlf := v1.ClientHeader.TlfName
```

Comment:
I thought both of them have the signature? @maxtaco mentioned earlier in a ticket that we need the authenticated `ClientHeader` because the body can be deleted.